### PR TITLE
Support pods in the `marathon deployment list` table view

### PR DESCRIPTION
**Pull request by cruhland**
- Add a new POD column to the table from "affectedPods" field
- Correctly handle the occurrence of "app" xor "pod" in actions

### CODE REVIEW

File: cli/tests/unit/data/deployment/app.txt [hunk 1]
```diff
@@ -0,0 +1,2 @@
+APP              POD  ACTION  PROGRESS  ID                                    
+/cassandra/dcos       scale     1/2     bebb8ffd-118e-4067-8fcb-d19e44126911
```
Comment (tamarrow) [inline cli/tests/unit/data/deployment/app.txt]: we should put a `-` under POD if it doesnt exist. same for app.

File: cli/dcoscli/tables.py [hunk 2]
```diff
@@ -145,9 +149,20 @@ def deployment_table(deployments):
 
     """
 
+    def resource_path_id(action):
+        path_id = action.get('app') or action.get('pod')
+
+        if path_id is None:
+            logger.exception('Missing app or pod entry')
```
Comment (tamarrow) [inline cli/dcoscli/tables.py]: lets log `action` too

File: cli/dcoscli/tables.py [hunk 3]
```diff
@@ -145,9 +149,20 @@ def deployment_table(deployments):
...
+
+            template = 'Expected "app" or "pod" field in action: {}'
+            raise ValueError(template.format(action))
```
Comment (tamarrow) [inline cli/dcoscli/tables.py]: instead of raising an exception, we should put 'N/A' under `ACTION`

File: cli/dcoscli/tables.py (hunk 2 above)
Comment (cruhland) [inline cli/dcoscli/tables.py]: You mean log if the `action` field does not exist?

File: cli/dcoscli/tables.py (hunk 3 above)
Comment (cruhland) [inline cli/dcoscli/tables.py]: So if any of the actions in a particular deployment have neither `"app"` nor `"pod"`, set the `ACTION` column for that row to `N/A`?

File: cli/tests/unit/data/deployment/app.txt (hunk 1 above)
Comment (cruhland) [inline cli/tests/unit/data/deployment/app.txt]: Ok, will do.

File: cli/dcoscli/tables.py (hunk 2 above)
Comment (tamarrow) [inline cli/dcoscli/tables.py]: log the json as part of the exception

File: cli/dcoscli/tables.py (hunk 3 above)
Comment (tamarrow) [inline cli/dcoscli/tables.py]: yep.

File: cli/dcoscli/tables.py [hunk 4]
```diff
@@ -149,21 +149,23 @@ def deployment_table(deployments):
 
     """
 
+    def join_path_ids(deployment, affected_resources_key):
```
Comment (tamarrow) [inline cli/dcoscli/tables.py]: pydoc

File: cli/dcoscli/tables.py [hunk 5]
```diff
@@ -149,21 +149,23 @@ def deployment_table(deployments):
...
+        path_ids = deployment.get(affected_resources_key)
+        return '\n'.join(path_ids) if path_ids else '-'
+
     def resource_path_id(action):
```
Comment (tamarrow) [inline cli/dcoscli/tables.py]: pydoc

File: cli/dcoscli/tables.py [hunk 6]
```diff
@@ -175,7 +177,9 @@ def get_action(deployment):
                 raise ValueError(
                     'Unknown Marathon action: {}'.format(action['action']))
 
-            if multiple_apps:
+            if None in action_path_ids:
```
Comment (tamarrow) [inline cli/dcoscli/tables.py]: should this be `if resource_path_id(action) is None`

**Comment by cruhland** [comment]
OS X tests were being flaky again.

**Pull request merged**